src/liballoc/vec.rs

```diff
@@ -2122,8 +2125,9 @@ where
         self.reserve(slice.len());
         unsafe {
             let len = self.len();
+            let dst_slice = slice::from_raw_parts_mut(self.as_mut_ptr().add(len), slice.len());
+            dst_slice.copy_from_slice(slice);
             self.set_len(len + slice.len());
-            self.get_unchecked_mut(len..).copy_from_slice(slice);
```

Comment:
Unfortunately there is no good way to grep for all places where the implicit `DerefMut` coercion is used, like here. I grepped for `get_unchecked_mut` and audited all of them, at least.

Reply:
Could you maybe print the backtrace in the `DerefMut` implementation and then run the tests and it will print what functions call `DerefMut`?

Reply:
Since `DerefMut` is called for every mutable indexing operation (`&mut v[i]`), I expect that to show tons of stacktraces and not be very useful.